module/x/gravity/keeper/signer_set.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/peggyjv/gravity-bridge/module/v3/x/gravity/types"
+)
+
+// TODO here we should check the contents of the validator set against
+// the store, if they differ we should take some action to indicate to the
+// user that bridge highjacking has occurred
+func (k Keeper) signerSetExecuted(ctx sdk.Context, nonce uint64) {
+	otx := k.GetOutgoingTx(ctx, types.MakeSignerSetTxKey(nonce))
+	if otx == nil {
+		k.Logger(ctx).Error("Failed to clean signer sets",
+			"signer set nonce", nonce)
+		return
+	}
+
+	completedSignerSetTx, _ := otx.(*types.SignerSetTx)
+	k.IterateOutgoingTxsByType(ctx, types.SignerSetTxPrefixByte, func(key []byte, otx types.OutgoingTx) bool {
```

Comment:
Is there a reason we're iterating over all the signer set TXs here? Isn't the pruning happening in ABCI?

Reply:
The pruning window is large and primarily for completed/timed out txs. This is mimicking other post-attestation handlers and immediately cleaning up any that can't be submitted because of nonce order. Is there an issue this creates that I'm missing?

Reply:
Signer sets can't really be canceled like the other TXs -- they aren't user originated, they are created by the chain in the order in which they need to be executed, sequentially. We should remove them individually as they complete rather than going through a list of old ones.